src/extension/wps/wps-core/src/main/java/org/geoserver/wps/ppio/ProcessParameterIO.java

```diff
@@ -184,8 +184,12 @@ public int compare(ProcessParameterIO o1, ProcessParameterIO o2) {
         }
 
         // load by factory
-        List<PPIOFactory> ppioFactories = GeoServerExtensions
-                .extensions(PPIOFactory.class, context);
+        List<PPIOFactory> ppioFactories;
+        if (context != null) {
+            ppioFactories = GeoServerExtensions.extensions(PPIOFactory.class, context);
```

Comment:
Are you talking about GeoServerExtensions.extensions? The warning happens here:

https://github.com/geoserver/geoserver-history/blob/master/src/platform/src/main/java/org/geoserver/platform/GeoServerExtensions.java#L296

Check how you get there, from what I see the context checked is null only if the caller did pass a null context, or if the caller did not pass a context _and_ GeoSErverExtensions bean has not yet been initialized. Generally speaking, beans depending on a context-less call should declare a dependency on GeoServerExtensions (not all do, not all can). Ideally, it would be nice if GeoServerExtensions was the very first bean initialized in the spring context. There is an article about this here:
http://rachitskillisaurus.blogspot.it/2013/10/spring-force-bean-to-be-first-to.html

Maybe the above is a better option than this fix, along with GeoServerExtensions automatically using its own default context if null is passed.